be/src/storage/delta_writer.cpp

```diff
@@ -339,16 +339,21 @@ Status DeltaWriter::commit() {
     return Status::OK();
 }
 
-void DeltaWriter::abort(bool with_log) {
+void DeltaWriter::abort(bool with_log, bool wait_flush) {
     _set_state(kAborted);
     _with_rollback_log = with_log;
+
+    if (wait_flush) {
+        // Any queued tasks not yet running are destroyed. If tasks are in flight, cancel() will wait
+        _flush_token->cancel();
```

Comment:
How about calling `_flush_token->cancel()` in the destructor of `DeltaWriter`

Reply:
> How about calling `_flush_token->cancel()` in the destructor of `DeltaWriter`

Good idea